src/standard.ts

```diff
@@ -150,6 +150,15 @@ export class StandardMerkleTree<T extends any[]> {
     return equalsBytes(impliedRoot, this.tree[0]!);
   }
 
+  static verifyWithRoot(leaf: any[], proof: string[], root: string, leafEncoding: string[]): boolean {
+    return StandardMerkleTree._verifyWithRoot(standardLeafHash(leaf, leafEncoding), proof.map(hexToBytes), root);
+  }
+
+  private static _verifyWithRoot(leafHash: Bytes, proof: Bytes[], root: string): boolean {
```

Comment:
What is this private function adding? I would just write all the code directly in `verifyWithRoot`.